### PR TITLE
move api to azdata

### PR DESCRIPTION
**Pull request by alanrenmsft**
1. as planned, moving the api definition from azdata.proposed to azdata
2. add notification about the deprecated module.

we are going to show a general message as warning prompt and log the detailed messages to the console in case there are multiple extensions with this problem.
user can disable the prompt by choosing ‘do not show again’
image
image

### CODE REVIEW

File: src/sql/workbench/api/browser/mainThreadExtensionManagement.ts [hunk 1]
```diff
@@ -29,4 +35,28 @@ export class MainThreadExtensionManagement implements MainThreadExtensionManagem
...
+
+	public $showObsoleteExtensionApiUsageNotification(message: string): void {
+		console.warn(message);
+
+		if (this._obsoleteExtensionApiUsageNotificationShown) {
+			return;
+		}
+
+		let enableObsoleteAPINotification = this._configurationService.getValue('workbench')['enableObsoleteApiUsageNotification'];
+		if (enableObsoleteAPINotification !== undefined && !enableObsoleteAPINotification) {
```
Comment (Charles-Gagnon) [inline src/sql/workbench/api/browser/mainThreadExtensionManagement.ts]: Why not just have it be neverShowObsoleteApiNotification and set that to true instead? Then you can simply just do

```
if (neverShowObsoleteApiNotification) {
…
```

Not a huge deal but it's easier to think about and is typically how I see config values like this be done

File: src/sql/workbench/api/browser/mainThreadExtensionManagement.ts [hunk 2]
```diff
@@ -29,4 +35,28 @@ export class MainThreadExtensionManagement implements MainThreadExtensionManagem
...
+			return;
+		}
+
+		this._notificationService.prompt(Severity.Warning,
+			localize('workbench.generalObsoleteApiNotification', "Some of the loaded extensions are using obsolete APIs, please find the detailed information in the console of Developer Tools"),
```
Comment (Charles-Gagnon) [inline src/sql/workbench/api/browser/mainThreadExtensionManagement.ts]: Hm, I feel like this reads a little oddly. Maybe something like 

```
"Some loaded extensions are using obsolete APIs, please find detailed information in the Developer Tools Console"
```

I'd also suggest making Developer Tools Console into a command-link to open the developer tools

File: src/sql/workbench/api/node/sqlExtHost.api.impl.ts [hunk 3]
```diff
@@ -562,6 +563,7 @@ export function createApiFactory(
 		// "sqlops" namespace provided for back-compat only, add new interfaces to "azdata"
 		sqlopsFactory: function (extension: IExtensionDescription): typeof sqlops {
 
+			extHostExtensionManagement.$showObsoleteExtensionApiUsageNotification(localize('ObsoleteApiModuleMessage', "The extension\"{0}\" is using sqlops module which has been replaced by azdata module, the sqlops module will be removed in a future release", extension.identifier.value));
```
Comment (Charles-Gagnon) [inline src/sql/workbench/api/node/sqlExtHost.api.impl.ts]: space after extension

**Comment by alanrenmsft** [comment]
added a open developer tools button

image

**Pull request merged**